Code/Mantid/Framework/PythonInterface/test/python/plugins/algorithms/DNSFlippingRatioCorrTest.py

```diff
@@ -0,0 +1,281 @@
+import unittest
+from testhelpers import run_algorithm
+from mantid.api import AnalysisDataService
+import numpy as np
+import mantid.simpleapi as api
+from mantid.simpleapi import DNSFlippingRatioCorr
+
+
+class DNSFlippingRatioCorrTest(unittest.TestCase):
+    workspaces = []
+    __sf_bkgrws = None
+    __nsf_bkgrws = None
+    __sf_nicrws = None
+    __nsf_nicrws = None
+
+    def _create_fake_workspace(self, wsname, dataY, flipper):
+        """
+        creates DNS workspace with fake data
+        """
+        ndet = 24
+        dataX = np.zeros(2*ndet)
+        dataX.fill(4.2 + 0.00001)
+        dataX[::2] -= 0.000002
+        dataE = np.sqrt(dataY)
+        # create workspace
+        api.CreateWorkspace(OutputWorkspace=wsname, DataX=dataX, DataY=dataY,
+                            DataE=dataE, NSpec=ndet, UnitX="Wavelength")
+        outws = api.mtd[wsname]
+        api.LoadInstrument(outws, InstrumentName='DNS')
```

Comment:
An alternative to avoid a potentially slow `LoadInstrument()` call in a unit test that needs at least a minimal instrument definition is to use the algorithm `CreateSampleWorkspace`. But that's probably not good enough for your test, as `CreateSampleWorkspace` creates a very simple fake instrument.
Let's keep an eye on how long this test takes to run, but since this is a Python test it will anyway take >=1-2s just for the interpreter startup etc. So I wouldn't be too concerned about having small loads here.